tests/tree/test_tree_node_parent.py

```diff
@@ -0,0 +1,10 @@
+from textual.widgets import TreeNode, Tree
+
+def test_tree_node_parent() -> None:
+    """It should be possible to access a TreeNode's parent."""
+    tree = Tree[None]("Anakin")
+    child = tree.root.add("Leia")
+    grandchild = child.add("Ben")
+    assert tree.root.parent is None
+    assert grandchild.parent == child
```

Comment:
(apologies if this in nit'ting, I don't mean it as such) I think it might be better to change this `==` and the one below to `is`. Basically I submit that this is the contract -- you get back a node that IS the parent, not just "looks the same". Or at least that's what I think the contract should be.